subsys/bluetooth/adv_prov/providers/fast_pair.c

```diff
@@ -41,6 +44,52 @@ int bt_le_adv_prov_fast_pair_set_battery_mode(enum bt_fast_pair_adv_battery_mode
 	return 0;
 }
 
+static enum bt_fast_pair_adv_mode get_adv_mode(bool pairing_mode, bool show_ui)
+{
+	enum bt_fast_pair_adv_mode adv_mode;
+
+	if (pairing_mode) {
+		adv_mode = BT_FAST_PAIR_ADV_MODE_DISCOVERABLE;
+	} else {
+		if (show_ui) {
+			adv_mode = BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_SHOW_UI_IND;
+		} else {
+			adv_mode = BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_HIDE_UI_IND;
+		}
+	}
+
+	return adv_mode;
+}
+
+static bool detect_discoverable_stop_condition(enum bt_fast_pair_adv_mode adv_mode,
+					       const struct bt_le_adv_prov_adv_state *state)
+{
+	static bool adv_session_rpa_rotated;
+
+	/* Reset the advertising session memory on not discoverable mode updates. */
+	if (adv_mode != BT_FAST_PAIR_ADV_MODE_DISCOVERABLE) {
+		adv_session_rpa_rotated = false;
+
+		return false;
+	}
+
+	if (state->new_adv_session) {
+		adv_session_rpa_rotated = false;
+
+		return false;
+	} else if (state->rpa_rotated || adv_session_rpa_rotated) {
```

Comment:
You can do `if` instead of `else if` as the earlier if ends with return.

Reply:
fixed